src/Block_Solver.jl

```diff
@@ -0,0 +1,190 @@
+const ipopt_statuses = Dict(
+  0 => :first_order,
+  1 => :acceptable,
+  2 => :infeasible,
+  3 => :small_step,
+  #4 => Diverging iterates
+  5 => :user,
+  #6 => Feasible point found
+  -1 => :max_iter,
+  #-2 => Restoration failed
+  #-3 => Error in step computation
+  -4 => :max_time, # Maximum cputime exceeded
+  -5 => :max_time, # Maximum walltime exceeded
+  #-10 => Not enough degress of freedom
+  #-11 => Invalid problem definition
+  #-12 => Invalid option
+  #-13 => Invalid number detected
+  -100 => :exception, # Unrecoverable exception
+  -101 => :exception, # NonIpopt exception thrown
+  -102 => :exception, # Insufficient memory
+  -199 => :exception, # Internal error
+)
+
+function block_grad!(nlp::MathOptNLPModel, x::AbstractVector, g::AbstractVector, λ::Float64, a::Float64)
```

Comment:
Oups. Here we just overlad the function provided in NLPModels.jl: 
```julia
function NLPModels.grad!(...)

```
No need to implement our own function as we are just overloading NLPModels.jl interface here

Reply:
Yeah, I should have done this.